### PR TITLE
[ready] Smooth-ish move rising: revengeance

### PR DESCRIPTION
**Pull request by actioninja**
## About The Pull Request

The least fucked up version to date(tm).

Also has the observer fix using a new signal based solution
https://www.youtube.com/watch?v=FUAmw5zdmYk

`glide_size` is now scaled based on current movespeed. This means 60 fps movement is far smoother and slow movement happens slower. The minimum speed is clamped to 5 because of byond awfulness.

Also lol you suckers are gonna be using 60 fps by default now.

## Changelog
:cl:
tweak: Glide size now scales with movespeed. 60 fps won't jitter your camera around and slower moving mobs will appear to actually be slower.
tweak: 60 fps is now the client default
/:cl:

### CODE REVIEW

**Comment by MrPerson** [comment]
I hope it works this time. It's hard to overstate how much nicer this makes everything feel.

**Comment by Tlaltecuhtli** [comment]
human eye cant see more than 30 fps

**Comment by PKPenguin321** [comment]
I've commented on this before but because sprites are our hitboxes this isn't actually cosmetic, it makes slower people significantly easier to click on and therefore hit because their sprites don't move as fast.

I think it's pretty lame since having slowdown is already crippling and doesn't need a buff, but I'll admit that this is a really cool effect

**Comment by tmtmtl30** [comment]
I really dislike this. Playing on the testmerge, it made everything feel weird and it also made hitting slow people quite a bit easier. It's absolutely trivial to hit somebody who's been knocked down.

**Comment by Rohesie** [comment]
Testing on TGMC, the main issue people have is forcing 60fps on everyone.

I haven't yet found a formula that correctly converts the move delay into a proper glide size so it's properly timed. The one in here doesn't do it, it stutters at some points.

**Comment by ninjanomnom** [comment]
The slower move animation making it easier to hit things is not something we care about.

**Comment by PepperPrepper** [comment]
(High ping player opinion) 
This makes the game look and feel **much** more smoother. I no longer feel like I'm lagging behind everyone. and it allows me to actually participate in melee combat. As it seems that the new smooth movement gives me a much higher chance to actually hit my target.

**Comment by Rohesie** [comment]
I addressed several pull and buckle interactions in https://github.com/tgstation/TerraGov-Marine-Corps/pull/3403 , in case you are interested.
Unfortunately it's mixed with a huge movement refactor to bring the movespeeds to /tg/ standards, and our buckles and pulls are not exactly the same (for now, at least).

**Review by actioninja — COMMENTED** [review]
possible concern:
It may be better to update existing player configs instead of changing what 0 fps means in options. This works, but 0 fps has always struck me as something kind of confusing and intuitive.

**Comment by AnturK** [comment]
I see a lot of infinite loops from this in the logs. From quick glance it's two people grabbing/pulling each other.

**Comment by actioninja** [comment]
Somehow I forgot that was possible. This part isn't needed because glide size is passed through via an override instead of set manually like this anyways.

File: code/datums/components/orbiter.dm [hunk 1]
```diff
@@ -16,6 +16,7 @@
 	orbiters = list()
 
 	begin_orbit(orbiter, radius, clockwise, rotation_speed, rotation_segments, pre_rotation)
+	RegisterSignal(parent, COMSIG_MOVABLE_UPDATE_GLIDE_SIZE, .proc/orbiter_glide_size_update)
```
Comment (ninjanomnom) [inline code/datums/components/orbiter.dm]: Move this to `RegisterWithParent()` and unregister in the sibling proc

File: code/modules/client/client_procs.dm [hunk 2]
```diff
@@ -249,7 +249,10 @@ GLOBAL_LIST_EMPTY(external_rsc_urls)
 		GLOB.preferences_datums[ckey] = prefs
 	prefs.last_ip = address				//these are gonna be used for banning
 	prefs.last_id = computer_id			//these are gonna be used for banning
-	fps = prefs.clientfps
+	if(prefs.clientfps)
+		fps = prefs.clientfps
+	else
+		fps = world.fps * 3 // 60 fps at 20 tps (tg default)
```
Comment (ninjanomnom) [inline code/modules/client/client_procs.dm]: Is there a reason this needs to be tied to world fps? If we for some reason changed world fps I don't see why the client fps should also change

File: code/modules/client/preferences.dm [hunk 3]
```diff
@@ -1465,10 +1465,13 @@ GLOBAL_LIST_EMPTY(preferences_datums)
...
 				if ("clientfps")
-					var/desiredfps = input(user, "Choose your desired fps. (0 = synced with server tick rate (currently:[world.fps]))", "Character Preference", clientfps)  as null|num
+					var/desiredfps = input(user, "Choose your desired fps. (0 = 3x server tickrate (currently:[world.fps * 3]))", "Character Preference", clientfps)  as null|num
 					if (!isnull(desiredfps))
 						clientfps = desiredfps
-						parent.fps = desiredfps
+						if(desiredfps)
+							parent.fps = desiredfps
+						else
+							parent.fps = world.fps * 3 // 60 fps at 20 tps (tg default)
```
Comment (ninjanomnom) [inline code/modules/client/preferences.dm]: Ditto

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: code/modules/client/client_procs.dm (hunk 2 above)
Comment (actioninja) [reply code/modules/client/client_procs.dm]: going to revert this part and just update player settings instead.

File: code/datums/components/orbiter.dm [hunk 4]
```diff
@@ -17,6 +17,12 @@
 
 	begin_orbit(orbiter, radius, clockwise, rotation_speed, rotation_segments, pre_rotation)
 
+/datum/component/orbiter/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_MOVABLE_UPDATE_GLIDE_SIZE, .proc/orbiter_glide_size_update)
```
Comment (ninjanomnom) [inline code/datums/components/orbiter.dm]: orbiter already has register and unregister code, just 10 lines down

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: code/modules/client/preferences_savefile.dm [hunk 5]
```diff
@@ -42,6 +42,10 @@ SAVEFILE UPDATING/VERSIONING - 'Simplified', or rather, more coder-friendly ~Car
 //if your savefile is 3 months out of date, then 'tough shit'.
 
 /datum/preferences/proc/update_preferences(current_version, savefile/S)
+	if(current_version < 30)
+		if(clientfps == 0)
+			clientfps = 60
+
 	if(current_version < 29)
```
Comment (ninjanomnom) [inline code/modules/client/preferences_savefile.dm]: The order should be old to new

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: code/datums/components/orbiter.dm (hunk 4 above)
Comment (actioninja) [reply code/datums/components/orbiter.dm]: *drools*

**Review by ninjanomnom — APPROVED** [review]

**Comment by MCterra10** [comment]
Aww ha ha, ahh that's hot, that's hot

**Comment by optimumtact** [comment]
this makes me quite sick moving as a ghost, and I have to up the fps to 120, at which point everyone jerks around like a muppet on a stick.

Plus, what's with the slow ass tile slide at the end?

2/10, apply yourselves

**Comment by carshalash** [comment]
yikes

**Comment by optimumtact** [comment]
just set your client fps back to whatever you had it before and it's honestly not different.

FYI, glide size doesn't scale with client fps anyway, it's been broken forever
http://www.byond.com/forum/post/2241289

so we should probably just go to pixel movement

**Comment by Rohesie** [comment]
I've tested this on TGMC before and warned about the results. It's not working as intended. The forumla is unable to properly take into account speed variations.

**Comment by Summoners** [comment]
Yeah this actually makes me get motion sick, especially as a ghost. I can't even look at the game in the current state.

**Comment by trollbreeder** [comment]
This just feels wrong to me.
Atleast make it an option if possible.

**Comment by ninjanomnom** [comment]
That's not possible

**Comment by optimumtact** [comment]
@Summoners @trollbreeder if you set your client fps back to 20 or 0 (match server fps) in your game prefs, it basically looks like it did previously.

**Comment by trollbreeder** [comment]
@optimumtact It does make it look like it did before, but only if you're duffel-bag wearing slow. The moment you take the duffel off and run, HUUUURK.

**Comment by optimumtact** [comment]
sad

**Pull request merged**